nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/scheduling/RepositoryContextFactory.java

```diff
@@ -54,7 +54,13 @@ public RepositoryContextFactory(final ContentRepository contentRepository, final
     }
 
     public RepositoryContext newProcessContext(final Connectable connectable, final AtomicLong connectionIndex) {
-        final StateManager stateManager = stateManagerProvider.getStateManager(connectable.getIdentifier());
+        final StateManager stateManager;
+        if (connectable instanceof org.apache.nifi.controller.ProcessorNode processorNode) {
```

Comment:
Is there a reason this is limited to Processors? Other components can also use state, and the annotation evaluation should not need to know about the type of class.